code/modules/shuttle/emergency.dm

```diff
@@ -411,21 +417,13 @@
 	return has_people && ((hijacker_count == 1) || (hijacker_count && !solo_hijack))
 
 /obj/docking_port/mobile/emergency/proc/is_hijacked_by_xenos()
-	var/has_xenos = FALSE
+	var/queen_alive_on_shuttle = FALSE
+	//checking all players
 	for(var/mob/living/player in GLOB.alive_mob_list)
-		if(issilicon(player)) //Borgs are technically dead anyways
-			continue
-		if(isanimal(player)) //animals don't count
-			continue
-		if(isbrain(player)) //also technically dead
-			continue
 		if(shuttle_areas[get_area(player)])
-			//Non-xeno present. Can't hijack.
-			if(!istype(player, /mob/living/carbon/alien))
-				return FALSE
-			has_xenos = TRUE
-
-	return has_xenos
+			if(isalienqueen(player))
+				queen_alive_on_shuttle = TRUE
```

Comment:
you can just return directly or break here

Reply:
👍